src/test/kotlin/org/wycliffeassociates/otter/common/domain/md/ParseMdTest.kt

```diff
@@ -0,0 +1,175 @@
+                            HelpResource("Title 4", ""),
+                            HelpResource("Title 5", "Body 5"),
+                            HelpResource("Title 6", "Body 6")
+                    )
+            )
+    )
+
+    // Testing title text extraction
+    private val testGetTitleTextCases = listOf(
+            mapOf(
+                    "line" to "# Hello",
+                    "expected" to "Hello"
+            ),
+            mapOf(
+                    "line" to "#  Matthew",
+                    "expected" to "Matthew"
+            ),
+            mapOf(
+                    "line" to "## John said",
+                    "expected" to "John said"
+            ),
+            mapOf(
+                    "line" to "# John said # hello",
+                    "expected" to "John said # hello"
+            ),
+            mapOf(
+                    "line" to "#John said hello",
+                    "expected" to "John said hello"
+            )
+    )
+
+    // Testing title recognition
+    private val testIsTitleLineCases = listOf(
+            mapOf(
+                    "line" to "# Matthew",
+                    "expected" to true
+            ),
+            mapOf(
+                    "line" to "## Matthew",
+                    "expected" to true
+            ),
+            mapOf(
+                    "line" to "",
+                    "expected" to false
+            ),
+            mapOf(
+                    "line" to "Matthew # said hello",
+                    "expected" to false
+            ),
+            mapOf(
+                    "line" to "# ",
+                    "expected" to false
+            ),
+            mapOf(
+                    "line" to "#Matthew",
+                    "expected" to true
+            )
+    )
+
+    private fun checkLineOperatorFunction(input: String, output: Any, expected: Any) {
+
+        try {
+            assertEquals(expected, output)
+        } catch (e: AssertionError) {
+            println("Input: $input")
+            println("Expected: $expected")
+            println("Result: $output")
+            throw e
+        }
+    }
+
+    @Test
+    fun testGetTitleText() {
+
+        testGetTitleTextCases.forEach {
+
+            val output = ParseMd.getTitleText(it["line"] as String)
+
+            checkLineOperatorFunction(it["line"] as String, output, it["expected"] as String)
+        }
+    }
+
+    @Test
+    fun testIsTitleLine() {
+
+        testIsTitleLineCases.forEach {
+
+            val output = ParseMd.isTitleLine(it["line"] as String)
+
+            checkLineOperatorFunction(it["line"] as String, output, it["expected"] as Boolean)
+        }
+    }
+
+    private fun getBufferedReader(lines: List<String>): BufferedReader {
+
+        val lineSeparator = System.lineSeparator()
+
+        val stream: ByteArrayInputStream = lines.joinToString(lineSeparator).byteInputStream()
+
+        return BufferedReader(stream.bufferedReader())
```

Comment:
I think using the BR constructor is redundant here.

Reply:
Good catch, maybe this was an artifact of an earlier failed attempt. This is one of those duh moments!